provider-example/src/aead.rs

```diff
@@ -4,21 +4,25 @@ use alloc::vec::Vec;
 use chacha20poly1305::aead::Buffer;
 use chacha20poly1305::{AeadInPlace, KeyInit, KeySizeUser};
 use rustls::crypto::cipher::{
-    self, AeadKey, BorrowedPayload, Iv, UnsupportedOperationError, NONCE_LEN,
+    make_tls12_aad, make_tls13_aad, AeadKey, BorrowedOpaqueMessage, BorrowedPayload,
+    InboundMessage, Iv, KeyBlockShape, MessageDecrypter, MessageEncrypter, Nonce, OpaqueMessage,
+    OutboundMessage, PrefixedPayload, Tls12AeadAlgorithm, Tls13AeadAlgorithm,
+    UnsupportedOperationError, NONCE_LEN,
 };
+use rustls::internal::msgs::codec::Codec;
 use rustls::{ConnectionTrafficSecrets, ContentType, ProtocolVersion};
 
 pub struct Chacha20Poly1305;
 
-impl cipher::Tls13AeadAlgorithm for Chacha20Poly1305 {
-    fn encrypter(&self, key: cipher::AeadKey, iv: cipher::Iv) -> Box<dyn cipher::MessageEncrypter> {
+impl Tls13AeadAlgorithm for Chacha20Poly1305 {
```

Comment:
I think it's a good idea to remove the type qualifiers here, but I would prefer to see that in a separate commit so that this one can focus on the actual logical changes.

Reply:
Done.